src/SqlPersistence/SynchronizedStorage/StorageAdapter.cs

```diff
@@ -23,19 +31,29 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(OutboxTransaction tr
             CompletableSynchronizedStorageSession session = new StorageSession(outboxTransaction.Connection, outboxTransaction.Transaction, false, infoCache);
             return Task.FromResult(session);
         }
-        return EmptyResult;
+        return EmptyResultTask;
     }
 
-    public Task<CompletableSynchronizedStorageSession> TryAdapt(TransportTransaction transportTransaction, ContextBag context)
+    public async Task<CompletableSynchronizedStorageSession> TryAdapt(TransportTransaction transportTransaction, ContextBag context)
     {
         SqlConnection existingSqlConnection;
         SqlTransaction existingSqlTransaction;
         //SQL server transport in native TX mode
         if (transportTransaction.TryGet(out existingSqlConnection) && transportTransaction.TryGet(out existingSqlTransaction))
         {
             CompletableSynchronizedStorageSession session = new StorageSession(existingSqlConnection, existingSqlTransaction, false, infoCache);
-            return Task.FromResult(session);
+            return session;
         }
+
+        Transaction existingTransaction;
+        // Transport supports DTC and uses TxScope owned by the transport
+        if (transportTransaction.TryGet(out existingTransaction))
```

Comment:
I think it should be fairly safe to do it this way since only SQL will add the connection and native tx to the transport transaction which should take precedence over this check here. Thoughts?